### PR TITLE
Bugfix infinite recursion RecursiveQuery lookupHost

### PR DESCRIPTION
**Pull request by kokizzu**
RecursiveQuery lookupHost RecursiveQuery lookupHost RecursiveQuery lookupHost being called again and again recursively, tested with this domain `globalcdn.co`

solution: add IncrementAttempt before starting recursion.

### CODE REVIEW

File: client/client.go [hunk 1]
```diff
@@ -106,7 +106,7 @@ func (c *Client) RecursiveQuery(m *dns.Msg, tracer Tracer) (r *dns.Msg, rtt time
 	qname := m.Question[0].Name
 	qtype := m.Question[0].Qtype
 	zone := "."
-	for i := 1; i < 100; i++ {
+	for z := 1; z < 4; z++ {
```
Comment (rs) [inline client/client.go:109]: Feel like this change is unrelated to this fix right?

Comment (kokizzu) [reply client/client.go:109]: ah yes, it just that the result the same either 100 or 4

Comment (kokizzu) [reply client/client.go]: reverted

Comment (rs) [reply client/client.go]: Can you please also revert the i->z rename?
